### PR TITLE
Updated p4rt-ctl.rst for direct counter Add, delete and get CLI

### PR DESCRIPTION
**Pull request by bharticemk**

### CODE REVIEW

File: docs/clients/p4rt-ctl.rst [hunk 1]
```diff
@@ -319,7 +319,61 @@ Details about the supported commands
     ``COUNTER_FLOW``: Refers to the counter ID (generated ID by p4c, refer to bfrt.json file)
     and the counter table index. Format "counter_id=<number>,index=<number>"
 
-16. Get flow dump entries ::
+
+16. Add a rule for direct counter into the forwarding pipeline::
```
Comment (nupuruttarwar) [inline docs/clients/p4rt-ctl.rst:323]: Please keep the format consistent, similar to line 259 - "Rule to program direct counter?

File: docs/clients/p4rt-ctl.rst [hunk 2]
```diff
@@ -319,7 +319,61 @@ Details about the supported commands
...
+
+    [Egress/Tx]
+    p4rt-ctl add-entry br0 my_control.e_fwd "hdrs.mac[vmeta.common.depth].da="0x000000000461",hdrs.mac[vmeta.common.depth].sa="0x9ebace98d9d3",action=my_control.ddcount_and_send(1)"
+
+  .. note::
+
+    ``SWITCH``: Refers to the bridge name, which maps to device name
+    internally.
+    ``TABLE``: Refers to table_name present in p4info.txt file.
+    ``FLOW``: Refers to parameters for a the above mentioned TABLE
```
Comment (nupuruttarwar) [inline docs/clients/p4rt-ctl.rst:338]: Remove "a" - for the above mentioned..

File: docs/clients/p4rt-ctl.rst [hunk 3]
```diff
@@ -319,7 +319,61 @@ Details about the supported commands
...
+    match_filed_key=value action=action_name(value)
+
+17. Delete a rule for direct counter from the forwarding pipeline::
```
Comment (nupuruttarwar) [inline docs/clients/p4rt-ctl.rst:341]: Rule to delete direct counter entry?

File: docs/clients/p4rt-ctl.rst [hunk 4]
```diff
@@ -319,7 +319,61 @@ Details about the supported commands
     ``COUNTER_FLOW``: Refers to the counter ID (generated ID by p4c, refer to bfrt.json file)
     and the counter table index. Format "counter_id=<number>,index=<number>"
 
-16. Get flow dump entries ::
+
+16. Add a rule for direct counter into the forwarding pipeline::
+
+    $ p4rt-ctl add-entry SWITCH TABLE FLOW
+    $ Example::
+    [Ingress/Rx]
```
Comment (nupuruttarwar) [inline docs/clients/p4rt-ctl.rst:327]: Please check rst formatting here:
image

Comment (nupuruttarwar) [reply docs/clients/p4rt-ctl.rst:327]: Same comment for all examples

File: docs/clients/p4rt-ctl.rst [hunk 5]
```diff
@@ -319,7 +319,61 @@ Details about the supported commands
     ``COUNTER_FLOW``: Refers to the counter ID (generated ID by p4c, refer to bfrt.json file)
     and the counter table index. Format "counter_id=<number>,index=<number>"
 
-16. Get flow dump entries ::
+
+16. Add a rule for direct counter into the forwarding pipeline::
+
+    $ p4rt-ctl add-entry SWITCH TABLE FLOW
```
Comment (5abeel) [inline docs/clients/p4rt-ctl.rst:325]: There are two other `p4rt-ctl add-entry` commands in this file. Why do we need 3 different versions of `add-entry` command documented?

**Comment by nupuruttarwar** [comment]
In examples, if you can keep the MAC address of Tx and Rx different, that will avoid confusion for the reader

File: docs/clients/p4rt-ctl.rst (hunk 3 above)
Comment (bharticemk) [reply docs/clients/p4rt-ctl.rst:341]: yes

File: docs/clients/p4rt-ctl.rst (hunk 5 above)
Comment (bharticemk) [reply docs/clients/p4rt-ctl.rst:325]: we need this "add-entry" for direct counter  and in CLI we are showing the same under action

File: docs/clients/p4rt-ctl.rst (hunk 1 above)
Comment (bharticemk) [reply docs/clients/p4rt-ctl.rst:323]: yes

File: docs/clients/p4rt-ctl.rst
Comment (ffoulkes) [inline docs/clients/p4rt-ctl.rst:1]: In the Details about Supported Commands section, I find the Note entry after each example distracting.

A Note is supposed to call the reader's attention to something that is especially important. The sections that are marked as notes aren't notable. They're ordinary explanations of the example above. They could just as easily be (and would normally be) simple paragraphs indented to align with the code block. They're body paragraphs under the numbered line.

```text
16. Swing your partner

   <code block>

   Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
   incididunt ut labore et dolore magna aliqua.

17. Allemande left
```

You might want to consider making them bulleted lists, with each definition being a bullet.

Also, is `match_filed_key` a typo for `match_field_key`? _filed_ seems inappropriate in this context.

File: docs/clients/p4rt-ctl.rst [hunk 6]
```diff
@@ -116,15 +116,15 @@ Details about the supported commands
...
 
-  .. note::
-
-    ``SWITCH``: Refers to the bridge name, which maps to device name
-    internally.
-    ``PROGRAM``: Refers to protobuf binary file generated by
-    tdi_pipeline_builder executable and the P4 program's configuration file.
-    ``P4INFO``: Refers to P4Info.txt file generated by p4c compiler
-    for a P4 program.
+   Parameter description:
```
Comment (ffoulkes) [inline docs/clients/p4rt-ctl.rst:121]: This section looks awful when rendered. I'd go back to the original version and try again some other time.

**Review by ffoulkes — APPROVED** [review]
LGTM

File: docs/clients/p4rt-ctl.rst (hunk 2 above)
Comment (bharticemk) [reply docs/clients/p4rt-ctl.rst:333]: Fixed

File: docs/clients/p4rt-ctl.rst (hunk 6 above)
Comment (bharticemk) [reply docs/clients/p4rt-ctl.rst]: Done

File: docs/clients/p4rt-ctl.rst (hunk 5 above)
Comment (bharticemk) [reply docs/clients/p4rt-ctl.rst]: Removed

**Pull request merged**